File: go/fiber/internal/modules/admin/service.go
package admin

import (
	"tcc/internal/common/errors"
	"tcc/internal/modules/client"
	c "tcc/internal/modules/client/entity"
	"tcc/internal/modules/manager"
	m "tcc/internal/modules/manager/entity"
)

type AdminService struct{
	repository struct{
		client client.ClientRepository
		manager manager.ManagerRepository
	}
}


func (s *AdminService) FindAllClients(quantidade int) ([]*c.ClientEntity, *errors.BaseError) {
	clients, _ := s.repository.client.FindAll(quantidade)

	if clients == nil {
		return []*c.ClientEntity{}, nil
	}
	return clients, nil
}

func (s *AdminService) FindAllManagers(quantidade int) ([]*m.ManagerEntity, *errors.BaseError) {
	managers, _ := s.repository.manager.FindAll(quantidade)

	if managers == nil {
		return []*m.ManagerEntity{}, nil
	}
	return managers, nil
}

File: rust/axum/src/modules/admin/service.rs
use crate::modules::manager::repository::ManagerRepository;
use crate::modules::client::repository::ClientRepository;
use crate::modules::client::entity::ClientEntity;
use crate::modules::manager::entity::ManagerEntity;
use crate::common::errors::base::BaseError;
use crate::common::errors::internal_server::internal_server_error;
use crate::common::messages::error::ErrorMessages;


pub struct AdminService {
}

impl AdminService{
    pub async fn find_all_clients(quantidade: i32) -> Result<Vec<ClientEntity>, BaseError> {
        let clients = ClientRepository::find_all(quantidade).await
            .map_err(|_| internal_server_error(ErrorMessages::INTERNAL_SERVER, None))?;

        if clients.is_empty() {
            return Ok(Vec::new());
        }

        Ok(clients)
    }

    pub async fn find_all_managers(quantidade: i32) -> Result<Vec<ManagerEntity>, BaseError> {
        let managers = ManagerRepository::find_all(quantidade).await
            .map_err(|_| internal_server_error(ErrorMessages::INTERNAL_SERVER, None))?;

        if managers.is_empty() {
            return Ok(Vec::new());
        }

        Ok(managers)
    }
}